sermant-backend/src/main/java/io/sermant/backend/dao/redis/RedisClusterClientImpl.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright (C) 2023-2023 Huawei Technologies Co., Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sermant.backend.dao.redis;
+
+import com.alibaba.fastjson.JSONObject;
+
+import io.sermant.backend.common.conf.BackendConfig;
+import io.sermant.backend.common.conf.CommonConst;
+import io.sermant.backend.dao.DatabaseType;
+import io.sermant.backend.dao.EventDao;
+import io.sermant.backend.entity.InstanceMeta;
+import io.sermant.backend.entity.event.Event;
+import io.sermant.backend.entity.event.EventsRequestEntity;
+import io.sermant.backend.entity.event.QueryCacheSizeEntity;
+import io.sermant.backend.entity.event.QueryResultEventInfoEntity;
+import io.sermant.backend.util.DbUtils;
+
+import org.apache.logging.log4j.util.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import redis.clients.jedis.CommandObjects;
+import redis.clients.jedis.Connection;
+import redis.clients.jedis.ConnectionPool;
+import redis.clients.jedis.ConnectionPoolConfig;
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.JedisClientConfig;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.resps.Tuple;
+
+/**
+ * Redis stand-alone client
+ *
+ * @author xuezechao
+ * @since 2023-03-02
+ */
+public class RedisClusterClientImpl implements EventDao {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RedisClusterClientImpl.class);
+
+    private final BackendConfig backendConfig;
+
+    private final CommandObjects commandObjects = new CommandObjects();
+
+    private JedisCluster jedisCluster;
+
+    /**
+     * Construct the Jedis connection pool
+     *
+     * @param backendConfig configuration
+     */
+    public RedisClusterClientImpl(BackendConfig backendConfig) {
+        this.backendConfig = backendConfig;
+        String[] addressList = backendConfig.getUrl().split(CommonConst.REDIS_CLUSTER_SPLIT);
```

Comment:
no need to process addressList again, put it into RedisClusterClientImpl constructor